test/src/test/java/jenkins/fingerprints/FingerprintStorageTest.java

```diff
@@ -0,0 +1,142 @@
+        assertThat(fingerprintLoaded, is(not(nullValue())));
+        assertThat(fingerprintSaved.toString(), is(equalTo(fingerprintLoaded.toString())));
+
+        // Ensure that the loaded fingerprint was deleted from local storage after being loaded.
+        fingerprintLoaded = ExtensionList.lookup(FingerprintStorage.class).get(FileFingerprintStorage.class).load(id);
+        assertThat(fingerprintLoaded, is(nullValue()));
+    }
+
+    @Test
+    public void testLoadingAndSavingFingerprintWithExternalStorage() throws IOException {
+        configureExternalStorage();
+        String id = Util.getDigestOf("testLoadingAndSavingFingerprintWithExternalStorage");
+        Fingerprint fingerprintSaved = new Fingerprint(null, "bar.jar", Util.fromHexString(id));
+        Fingerprint fingerprintLoaded = ExtensionList.lookup(FingerprintStorage.class).get(TestExternalFingerprintStorage.class).load(id);
+        assertThat(fingerprintLoaded, is(not(nullValue())));
+        assertThat(fingerprintSaved.toString(), is(equalTo(fingerprintLoaded.toString())));
+    }
+
+    @Test
+    public void testDeletingLocalStorageFingerprintWithExternalStorageBeforeMigration() throws IOException {
+        String id = Util.getDigestOf("testLoadingAndSavingLocalStorageFingerprintWithExternalStorage");
+        new Fingerprint(null, "foo.jar", Util.fromHexString(id));
+        configureExternalStorage();
+        Fingerprint.delete(id);
+        Fingerprint fingerprintLoaded = ExtensionList.lookup(FingerprintStorage.class).get(FileFingerprintStorage.class).load(id);
+        assertThat(fingerprintLoaded, is(nullValue()));
+    }
+
+    @Test
+    public void testDeletingLocalStorageFingerprintWithExternalStorageAfterMigration() throws IOException {
+        String id = Util.getDigestOf("testLoadingAndSavingLocalStorageFingerprintWithExternalStorage");
+        new Fingerprint(null, "foo.jar", Util.fromHexString(id));
+        configureExternalStorage();
+        Fingerprint.load(id);
+        Fingerprint.delete(id);
+        Fingerprint fingerprintLoaded = ExtensionList.lookup(FingerprintStorage.class).get(TestExternalFingerprintStorage.class).load(id);
+        assertThat(fingerprintLoaded, is(nullValue()));
+    }
+
+    @Test
+    public void testDeletingFingerprintWithExternalStorage() throws IOException {
+        configureExternalStorage();
+        String id = Util.getDigestOf("testLoadingAndSavingLocalStorageFingerprintWithExternalStorage");
+        configureExternalStorage();
+        new Fingerprint(null, "foo.jar", Util.fromHexString(id));
+        Fingerprint.delete(id);
+        Fingerprint fingerprintLoaded = Fingerprint.load(id);
+        assertThat(fingerprintLoaded, is(nullValue()));
+    }
+
+    private void configureExternalStorage() {
+        ExtensionList.lookup(FingerprintStorage.class).add(0, new TestExternalFingerprintStorage());
+    }
+
+    public static class TestExternalFingerprintStorage extends FingerprintStorage {
```

Comment:
Shouldn't this be private? I'm curious why not have a mock for this object.

Reply:
I want to re-use this class in another test also. Would you suggest creating this in a separate file? I think that would be better.
So would you suggest using something like Mockito for this?

Reply:
I think using Mockito might be better (you get coverage on the actual class instead of your test class) and lighter-weight, no?

Reply:
The only problem I think is that I want to test the functionality too. IIUC Mockito will give me a placeholder, but it cannot act as an external storage, which I want to test if it is working or not.